cloud/ec2.go

```diff
@@ -1214,6 +1214,11 @@ func (m *ec2Manager) StartInstance(ctx context.Context, h *host.Host, user strin
 			if ec2StatusToEvergreenStatus(*instance.State.Name) == StatusRunning {
 				return false, nil
 			}
+			if h.IPv4 == "" {
```

Comment:
you shouldn't just do this if h.IPv4 is empty; it's possible that we stopped the instance and then restarted it, in which case the IPv4 address will have changed. Also right now you're only doing this for the "not running" case which is the opposite of what you want.

Reply:
makes sense! done